hooks/pre/30-el8_upgrade_postgresql.rb

```diff
@@ -0,0 +1,26 @@
+def postgresql_upgrade(new_version)
+  stop_services
+
+  execute!("dnf module switch-to postgresql:#{new_version} -y")
+
+  server_packages = ['postgresql', 'postgresql-server', 'postgresql-upgrade']
+  ['postgresql-contrib', 'postgresql-docs'].each do |extra_package|
+    if execute("rpm -q #{extra_package}", false, false)
+      server_packages << extra_package
+    end
+  end
+
+  ensure_packages(server_packages, 'latest')
+
+  execute!("sed -i '/^data_directory/d' /var/lib/pgsql/data/postgresql.conf")
```

Comment:
This is needed because `puppetlabs-postgresql` sets `data_directory` in the config, but `postgresql-setup` complains:
```
# su - postgres -c 'postgresql-setup --upgrade'
ERROR: data_directory field in configuration file is not supported.
       db datadir (PGDATA) needs to be specified exclusively in service/unit
       file as an Environment variable.
       In order to use this script, please remove data_directory entry from
       configuration file and make sure that the default location
       (PGDATA) in .service file is valid.
```

Reply:
Could we reference this in the line somehow?